samza-core/src/main/java/org/apache/samza/operators/triggers/RepeatingTriggerImpl.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.operators.triggers;
+
+import org.apache.samza.operators.data.MessageEnvelope;
+
+/**
+ * Implementation class for a {@link RepeatingTrigger}
+ */
+public class RepeatingTriggerImpl<M extends MessageEnvelope> implements TriggerImpl<M> {
+
+  private final Trigger<M> underlyingTrigger;
+
+  private TriggerImpl<M> underlyingTriggerImpl;
```

Comment:
It seems to me that it would be easier to read if we keep private final RepeatingTrigger<M> instead of underlyingTrigger here. And I believe underlyingTriggerImpl is just referring to the current TriggerImpl object for the repeating trigger, right? If yes, can we rename it to currentTriggerImpl?